src/Account/TSAccountManager.m

```diff
@@ -260,6 +270,11 @@ + (void)unregisterTextSecureWithSuccess:(void (^)())success failure:(void (^)(NS
         success:^(NSURLSessionDataTask *task, id responseObject) {
             DDLogInfo(@"%@ Successfully unregistered", self.tag);
             success();
+
+            // The success handler should reset local storage.
```

Comment:
> it seems like this is a TODO rather than a documentation of current behavior. 

Nope, this document current behavior.  This is called from `[SettingsTableViewController proceedToUnregistration]` whose success handler calls `[Environment resetAppData]`.   This method, after calling that success handler, then fires `kNSNotificationName_RegistrationStateDidChange` which is only safe to fire after the data store is reset.  This isn't a great arrangement, but I can see why we landed here (SSK doesn't know about `Environment`, etc.).  Hence the comment.  

I've elaborated the comment to more clear.